test/bin/s3-server.js

```diff
@@ -73,55 +83,101 @@ function createTempDirForBucket() {
   return tmpDirBucketPath;
 }
 
-/* eslint-disable max-nested-callbacks */
-fs.stat(path, (err, stats) => {
-  if (err || !stats.isDirectory()) {
-    process.exit(1);
-  }
-  const bucket = getBucketName(path);
-  const tmpDir = createTempDirForBucket();
-
-  const client = new S3rver({
-    port,
-    hostname,
-    silent: false,
-    directory: tmpDir
+/**
+ * Copies files from the watched folder the the "bucket"
+ * @param {String} bucket
+ */
+function copyFiles(bucket) {
+  const walker = walk.walk(path);
+
+  walker.on('file', (root, fileStats, next) => {
+    const pathToFile = `${Path.relative(path, root)}/${fileStats.name}`;
+    const stream = fs.createReadStream(`${path}/${pathToFile}`);
+
+    awsClient.putObject({
+      Bucket: bucket,
+      Key: pathToFile,
+      Body: stream
+    }, (err) => {
+      if (err) {
+        Log.info(err, err.stack);
+      }
+      next();
+    });
   });
+}
 
-  client.run((serverErr, host, p) => {
-    if (serverErr) {
-      process.exit(1);
+/**
+ * Deletes the contents of the "bucket"
+ * @param {String} bucket
+ */
+function emptyBucket(bucket) {
+  awsClient.listObjects({Bucket: bucket}, (listErr, listData) => {
+    if (listErr) {
+      throw listErr;
     }
-
-    awsClient.createBucket({Bucket: bucket}, (createBucketErr) => {
-      if (createBucketErr) {
-        throw createBucketErr;
-      }
-
-      const walker = walk.walk(path);
-
-      walker.on('file', (root, fileStats, next) => {
-        awsClient.putObject({Bucket: bucket, Key: `${Path.relative(path, root)}/${fileStats.name}`}, (putObjectErr) => {
-          if (putObjectErr) {
-            console.log(err, err.stack); // eslint-disable-line no-console
-          }
-          next();
-        });
+    listData.Contents.forEach((el) => {
+      awsClient.deleteObject({Bucket: bucket, Key: el.Key}, (deleteErr) => {
+        if (deleteErr) {
+          throw deleteErr;
+        }
       });
     });
-
-    console.log(`listening for S3 requests at http://${host}:${p}`); // eslint-disable-line no-console
   });
-});
-
-/* eslint-enable max-nested-callbacks */
+}
 
+/**
+ * Handles cleanup on app exit
+ */
 function exitHandler() {
-  console.log('Cleaning up temp directory'); // eslint-disable-line no-console
+  Log.info('Cleaning up temp directory');
   cleanupTempDir();
   process.exit(0);
 }
 
```

Comment:
Do we need API docs for the `createBucket` function so it matches up with the `emptyBucket` function?